optuna/samplers/_tpe/multi_objective_sampler.py

```diff
@@ -0,0 +1,628 @@
+import math
+from typing import Any
+from typing import Callable
+from typing import cast
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import numpy as np
+
+import optuna
+from optuna import distributions
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.samplers._random import RandomSampler
+from optuna.samplers._tpe.parzen_estimator import _ParzenEstimator
+from optuna.samplers._tpe.parzen_estimator import _ParzenEstimatorParameters
+from optuna.samplers._tpe.sampler import TPESampler
+from optuna.study import StudyDirection
+
+
+EPS = 1e-12
+_SPLITCACHE_KEY = "motpe:splitcache"
+_WEIGHTS_BELOW_KEY = "motpe:weights_below"
+
+
+def default_gamma(x: int) -> int:
+    return int(np.floor(0.1 * x))
+
+
+def _default_weights_above(x: int) -> np.ndarray:
+    return np.ones(x)
+
+
+@experimental("2.3.0")
```

Comment:
This version should be bumped to match our upcoming release.
```suggestion
@experimental("2.4.0")
```